docs/client-specifications/configuration.md

```diff
@@ -0,0 +1,147 @@
+ */
+const prodConfig = config.ProdLowLatency.latest();
+```
+
+## Configuration interface
+
+The configuration interface provides nested objects for retry strategy, middleware, and transport:
+
+```typescript
+interface Configuration {
+  retryStrategy: RetryStrategy,
+  middleware: Middleware,
+  transportStrategy: TransportStrategy
+}
+```
+
+### RetryStrategy
+
+The `RetryStrategy` provides a function that is called when a request fails.  This function is responsible for determining
+whether the request should be retried, and if so, how long the client should delay before retrying:
+
+```typescript
+interface RetryStrategy {
+  /**
+   * Returns the number of milliseconds after which the request should be retried, or `undefined` if the request should
+   * not be retried.  `0` means retry immediately.
+   */
+  determineWhenToRetryRequest(grpcResult: GrpcResult, grpcRequest: GrpcRequest, attemptNumber: number): Promise<number | undefined>;
+}
+```
+
+The simplest default implementation of this strategy can expose a `maxAttempts` integer, and always return `0` for
+immediate retries until the maximum number of attempts is reached.  A slightly more sophisticated initial implementation
+would have basic support for exponential backoff and jitter.
+
+### Middleware
+
+The `Middleware` interface allows the Configuration to provide a higher-order function that wraps all requests.  This
+allows future support for things like client-side metrics or other diagnostics helpers.
+
+```typescript
+interface Middleware {
+  wrapRequest(middlewareFn: (GrpcRequest) => GrpcResult): GrpcResult;
+}
+```
+
+### TransportStrategy
+
+The `TransportStrategy` is responsible for configuring network tunables.  We intend to abstract gRPC away from
+users as much as possible, but we will need to provide different gRPC configurations for dev vs. prod environments.
+And ultimately we will need to support advanced customers who need to mess with these knobs.
+
+```typescript
+interface TransportStrategy {
+  grpcConfig: GrpcConfiguration
+}
+
+interface GrpcConfiguration {
+  numChannels(): number
+
+  // These interface functions expose individual settings for gRPC channels.  They are just here to ensure that
+  // strategy implementations provide values for settings that we know to be important.  These may vary by language
+  // since the gRPC implementations in each language have subtly different behaviors.
+  maxSessionMemory(): number
+  useLocalSubChannelPool(): boolean
+
+  // This is a map that encapsulates the settings above, and may also include other channel-specific settings.
+  // This allows strategy implementations to provide gRPC config key/value pairs for any available setting, even
+  // if it's not one we've explicitly tried / recommended.  The strategy implementation should implement this by
+  // calling the functions above, along with allowing a mechanism for specifying additional key/value pairs.
+  grpcChannelConfig(): Record<string, string>
```

Comment:
so as end-user would I pass a  config like this:
```typescript
const client = new SimpleCacheClient({
  config: {
    numChannels: 4,
    grpcChannelConfig: {
      'numChannels': '6'
      'ignoreUnknownServices': 'true'
    }
  }
})
```

Obviously this is weird use case passing `numChannels` twice but would this work from end user perspective? They would need to pass each overide as strings? Would overides in `grpcChannelConfig` take precedent higher up fields? 

Maybe just add a small section on using this for advanced usage overrides with an example or 2?

Reply:
Yeah that's not exactly what I have in mind, but I can provide an advanced usage example, that's a fair request.

Reply:
added some examples in https://github.com/momentohq/standards-and-practices/pull/14/commits/7c54ee1016b39bef24280ead497e121de398e974